crates/iota/src/client_commands.rs

```diff
@@ -3047,3 +3051,33 @@ fn parse_emit_option(s: &str) -> Result<HashSet<EmitOption>, String> {
 
     Ok(options)
 }
+
+/// Warn the user if the CLI falls behind the current on-chain protocol
+/// version.
+async fn check_protocol_version_and_warn(client: &IotaClient) -> Result<(), anyhow::Error> {
+    let current_protocol_version = client
+        .read_api()
+        .get_protocol_config(None)
+        .await?
+        .protocol_version
+        .as_u64();
+
+    let build_protocol_version = ProtocolVersion::MAX.as_u64();
+    if build_protocol_version != current_protocol_version {
```

Comment:
IMO both `build_` and `current_` are confusing names. `current_` could apply to both the CLI and the network. `build_` is just weird? I'd prefer the upstream names of `cli_` and `on_chain_`, it's explicit.

Reply:
okidoki

Reply:
Let's see what the others think, the product being used is the CLI, we don't really care about the libraries at this point

Reply:
changed it back: https://github.com/iotaledger/iota/pull/5042/commits/82030a821e4b5374b3a38a64339ad180c7789a29